### PR TITLE
Added or expanded 5 comments…

### PR DESCRIPTION
**Pull request by liamzee**
…I documented lightly defaultMainHelper in Cabal/Distribution.Simple, expandResponse in Cabal/Distribution.Compat.ResponseFile, expanding the main comment in cabal-install/Distribution.Client.Main, adding segmentation commenting marking responseFiles handling in cabal-install/Distribution.Client.Main (main), added haddock documentation to mainWorker in cabal-install/Distribution.Client.Main .

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#other-conventions).

### CODE REVIEW

**Comment by ffaf1** [comment]
@liamzee , I see closed PRs with similar name, remember you can open one, wait for CI to finish and then update it (push with `--force`).

Is `fourmolu` pestering you?  Check [this bit](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#code-style) of `CONTRIBUTING.md`.

If you need more help, just ask!

**Comment by liamzee** [comment]
I know the implicit issue, i.e, some maintainers have e-mail alerts set to any attempt at PR, and  I just spammed multiple people's inboxes, and I'm really sorry about that.

make style is apparently going berserk, but I'm not going to ask for people's time when I can't even get my fourmolu working on VSC (It's apparently either ignoring the fourmolu.yaml or just diverting to a different piece of software).

Hope that something at least useful comes out of this PR.

**Comment by ffaf1** [comment]
> I know the implicit issue, i.e, some maintainers have e-mail alerts set to any attempt at PR, and I just spammed multiple people's inboxes, and I'm really sorry about that.

Contributions are always appreciated, helping each other is paramount!

`fourmolu` CI check passed, well done.

File: cabal-install/src/Distribution/Client/Main.hs
Comment (ffaf1) [inline cabal-install/src/Distribution/Client/Main.hs:318]: Curious, is there a reason why you keep your comment lines way shorter than the 80max customary ?limit?

File: Cabal/src/Distribution/Compat/ResponseFile.hs
Comment (ffaf1) [inline Cabal/src/Distribution/Compat/ResponseFile.hs:72]:
```
f@x270:~$ cabal @foo @foo
"?"
"?"
Error: cabal: unrecognised command: @foo (try --help)
Maybe you meant `info`?
```

How is this related to line-lenght limitations?

File: Cabal/src/Distribution/Simple.hs
Comment (ffaf1) [inline Cabal/src/Distribution/Simple.hs:158]: What is a “splitter”, in thix context?

File: Cabal/src/Distribution/Compat/ResponseFile.hs
Comment (geekosaur) [reply Cabal/src/Distribution/Compat/ResponseFile.hs:72]: That should probably be documented for people who don't know about response files.

**Comment by ffaf1** [comment]
Commit message should be formatted to [standard way](https://cbea.ms/git-commit/#seven-rules), as per `CONTRIBUTING.md`.

File: cabal-install/src/Distribution/Client/Main.hs
Comment (liamzee) [reply cabal-install/src/Distribution/Client/Main.hs:318]: I'm attempting to improve readability by verticalizing comments, within reason. The goal would be for someone viewing the comments manually to not have to scan too far for the entire semantic unit, with the newlines providing an obvious space to pause.

A further issue is, say, disambiguation. Cabal/Distribution.Simple naturally would want a period at the end, but is the period attached to the module name or the sentence?

I have no issue with editing, if this is seen to be an issue.

**Comment by liamzee** [comment]
What would you recommend as a path of action, then?

Step 1. Edit the commit message to match requirements; i.e, split the subject line with a summary,
then specify what occurred.

Step 2. Expand some of the comments to match line width expectations for comments.

Step 3. Be clearer on what a splitter is, or rather, remove the term entirely. Instead,
specify that it is the chooser for Cabal's simple build system to take an action.

Step 4. I am personally dissatisfied with one comment (mainHelper), as I don't think it's clear enough
on what one of the actions does. I was originally planning to revise the comment in a further
commit, but if I must revise, then it makes sense to make such changes immediately.

**Comment by ffaf1** [comment]
For sure, @liamzee!

**Review by geekosaur — APPROVED** [review]

**Review by ffaf1 — APPROVED** [review]
Very good!

**Comment by ffaf1** [comment]
Add a `merge me` and wait patiently for two days to have this merged, @liamzee.

**Pull request merged**
